### ISSUE TITLE
no scrolling graph when in realtime

### ISSUE THREAD

**Issue by massdonati** [FlowForwarding/dynamiccharts#3]
how can i tell the chart not to spread itself on the view since the first points but do it gradually, in a way that the more data I have to display the bigger width the graph has?!
NCISimpleGraphView has a xStep property that It should be doing that: setting the distance in between two points. Is that right? and if it's right how come that is not working?

**Comment by dostanko** [comment]
no, xStep  is scale  (difference between arguments of nearest points)

but again NCISimpleGraphView  - is not for public usage, use NCISimpleChartView and nciXLabelsDistance for your purposes

**Comment by massdonati** [comment]
So I could manage to realize the realtime chart by adding points and at some point to removeObjectAtIndex from the chartData Mutable array.. still not sure what xStep does and how do I make the x axe labels slide to the left as the graph slide to the left instead of changing the text of the labels.

**Comment by dostanko** [comment]
You should not use xStep, and should not create  NCISimpleGraphView instance - it's for internal library usage.

Please use  NCISimpleChartView

CHART instead GRAPH in class name.

**Comment by dostanko** [comment]
I guess work around you need is

```
NCISimpleChartView *chart = [[NCISimpleChartView alloc]
                             initWithFrame:CGRectMake(50, 30, 400, 250)];
[self.view addSubview:chart];
[chart addPoint:0 val:@[[NSNull null]]];
[chart addPoint:19 val:@[[NSNull null]]];

dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
    for (int ind = 0; ind < 19; ind ++){
        [NSThread sleepForTimeInterval:2.0f];
        [chart.chartData removeLastObject];
        [chart addPoint:ind val:@[@(arc4random() % 5)]];
        [chart addPoint:19 val:@[[NSNull null]]];
        dispatch_async(dispatch_get_main_queue(), ^{
            [chart drawChart];
        });
    }
});
```

**Comment by massdonati** [comment]
well that code is just to add values in real time (ad it works very well actually) what I was trying to achieve was about the x labels, I'd like them to follow the data in the chart like if the were static on a roller.

I that case the values at second 10 (and by values i mean the graph Ys and the x lable)they would be at second 10 but at second 11 the would be shifted of one position on the left and so on

**Comment by dostanko** [comment]
I can't understand what you need.
Could you please provide video or several consecutive images to describe it?
Please pay attention there was fixes release 1.0.3. May be it can help you.

Also I wrote you a letter here mass.donati@gmail.com for the case if you can  clarify your requirements via Skype call

**Comment by dostanko** [comment]
I think I've gotten you need, please check the code below

```
NCISimpleChartView *chart = [[NCISimpleChartView alloc]
                             initWithFrame:CGRectMake(50, 30, 400, 250)];
for (int ind = 0; ind < 10; ind ++){
    [chart addPoint:ind val:@[@(arc4random() % 5)]];
}
[self.view addSubview:chart];

dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
    for (int ind = 10; ind < 30; ind ++){
        [NSThread sleepForTimeInterval:1.0f];
        [chart.chartData removeObjectAtIndex:0];
        [chart addPoint:ind val:@[@(arc4random() % 5)]];
        dispatch_async(dispatch_get_main_queue(), ^{
            [chart drawChart];
        });
    }
});
```
